### ISSUE TITLE
inner async function without return value triggers S2259

### ISSUE THREAD

**Issue by vogsphar** [SonarSource/SonarJS#1116]
**I want to report a bug.**

**SonarJS version:** 5.0 (build 6962) <br>
**SonarQube version:** <br>
**SonarLint version:** <br>

**Rule key:** javascript:S2259

**may be related** #895 #732 #727

**Reproducer** 
```
function persist () {
  async function init () {
    console.log('init')
  }
  init().then(x => console.log(typeof x))
}
persist()
```

**Logs**
```
Properties of variables with "null" or "undefined" values should not be accessed
```

**Expected behavior**

No error is reported as `init` is defined here and it returns a promise as it is a async function.

**Workaround**
add a return value to the async function. According to Spec a return value is not needed for async functions. i.e. see examples at https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/async_function

**Comment by vogsphar** [comment]
`return null` -> S2259
`return undefined` -> S2259
`return 0` -> ok

**Comment by vilchik-elena** [comment]
In symbolic execution when checking for implicit return we should check if function is `async`:
https://github.com/SonarSource/SonarJS/blob/master/javascript-frontend/src/main/java/org/sonar/javascript/se/SymbolicExecution.java#L325
